VideoApp/VideoApp/Stores/Auth/CommunityAuthStore.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  Copyright (C) 2019 Twilio, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+class CommunityAuthStore: AuthStoreWriting {
+    weak var delegate: AuthStoreWritingDelegate?
+    var isSignedIn: Bool { keychainStore.passcode != nil }
+    var passcode: String? { keychainStore.passcode }
+    var userDisplayName: String { appSettingsStore.userIdentity }
+    private let api: APIConfiguring & APIRequesting
+    private let appSettingsStore: AppSettingsStoreWriting
+    private let keychainStore: KeychainStoreWriting
+
+    init(
+        api: APIConfiguring & APIRequesting,
+        appSettingsStore: AppSettingsStoreWriting,
+        keychainStore: KeychainStoreWriting
+    ) {
+        self.api = api
+        self.appSettingsStore = appSettingsStore
+        self.keychainStore = keychainStore
+    }
+
+    func start() {
+        guard let passcode = keychainStore.passcode else { return }
+        
+        configureAPI(passcode: passcode)
```

Comment:
When user is already signed in at app launch the API client needs to be configured so that when it is used it works. We persist the passcode in keychain.